w3d_translator/w3d_translator.py

```diff
@@ -46,59 +39,42 @@ def farewell():
     exit(0)
 
 
-# Farewell message on error
-def farewell_error():
+# Run doctests
+def run_tests():
+    typer.echo("Running Tests")
+    import translate as module
+
+    doctest.testmod(module)
     typer.echo()
-    typer.echo("Translation Completed with errors")
-    typer.echo("Please see the command line for a list of issues")
-    exit(1)
 
 
 # Translate a single project
 def translate_project(project_dir: Path, out_dir: Path, dev: bool = False):
     try:
         typer.echo(f"Translating project:\t {cyan(project_dir.name)}")
         validate_project(project_dir)
-
-        # Create Unity project and copy original files
         unity_dir = Path(out_dir, project_dir.name)
+
+        # Create Unity project
         if not dev:
-            create_project(unity_dir)
-            copy_files(project_dir, unity_dir)
-            add_empty_scene(unity_dir)
+            create_project(project_dir, unity_dir)
 
-        # Translate .xml files to .unity files
+        # Translate .xml files to .unity files (skip invalid)
         xml_files = [
             p
             for p in project_dir.iterdir()
             if (p.is_file() and p.suffix == ".xml")
         ]
-        schema = etree.XMLSchema(file="schema/caveschema.xsd")
         for file in xml_files:
             typer.echo(f"Translating file:\t {green(file.name)}")
-
-            # Validate xml file against schema file
             try:
-                schema.assertValid(etree.parse(file))
-            except etree.DocumentInvalid as e:
-                typer.echo(
-                    red(f"{file.name} does not match schema:"), err=True
-                )
-                # TODO: Shorten up error message, just line and error (30)
-                for error in e.error_log:
-                    typer.echo(red(error), err=True)
-                typer.echo(red(f"Skipping {file.name}"), err=True)
-            except Exception as e:
-                typer.echo(red(f"Error validating {file.name}"), err=True)
+                validate_xml(file)
+            except XmlError as e:
                 typer.echo(red(e), err=True)
-                typer.echo(red(f"Skipping {file.name}"), err=True)
             else:
-                # Build Story dataclass and Unity project
                 story = generateDS.parse(file, silence=True)
-                print(story.member_data_items_.keys())
                 build_project(unity_dir, story)
-
-    except (ValidationError, UnityError, TranslationError) as e:
+    except (ValidationError, UnityError) as e:
```

Comment:
is there any chance that there are other errors? do we need another generic except block?

Reply:
Nothing that shouldn't crash the program anyways I don't think - especially while I'm still developing things. I'd rather python just error out and debug from there than catch anything at this stage